src/components/sectioned-navigation/sectioned-navigation.js

```diff
@@ -16,12 +24,55 @@ class SectionedNavigation extends React.Component {
     );
   }
 
+  renderFilterBar() {
+    const { props } = this;
+    if (!props.includeFilterBar) {
+      return null;
+    }
+    return (
+      <div className="wmax360 mb24">
+        <input
+          onChange={e =>
+            this.setState({ filter: e.target.value }, this.filterResults())
+          }
+          type="text"
+          className="px6 py6 w-full round border border--gray-light"
+          name="filter"
+          placeholder="Filter examples"
+        />
+      </div>
+    );
+  }
+
+  filterResults() {
+    const filter = this.state.filter.toLowerCase().trim();
+    const visibleSections = this.props.sections
+      .filter(section => {
+        const matchedItems = section.items.filter(item => {
+          return item.text.toLowerCase().indexOf(filter) > -1;
+        });
+        return matchedItems.length > 0;
+      })
+      .map(filteredSection => {
+        const filteredItems = filteredSection.items.filter(item => {
+          return item.text.toLowerCase().indexOf(filter) > -1;
+        });
+        return {
+          title: filteredSection.title,
+          url: filteredSection.url,
+          items: filteredItems
+        };
+      });
```

Comment:
Nice attention to the details here, keeping sections open and filtering their items.